base/database.py

```diff
@@ -1577,21 +1615,43 @@ def tag(ea, key, none):
     # if not within a function, then fetch the repeatable comment otherwise update the non-repeatable one
     try:
         func = function.by_address(ea)
+        rt, _ = interface.addressOfRuntimeOrStatic(func)
     except E.FunctionNotFoundError:
-        func = None
-    repeatable = False if func and function.within(ea) else True
+        rt, func = False, None
+    repeatable = False if func and function.within(ea) and not rt else True
+
+    # figure out which comment type the user's key is in so that we can remove
+    # that one. if we're a runtime-linked address, then we need to remove the
+    # tag from a repeatable function comment. if the tag isn't in any of them,
+    # then it doesn't really matter since we're going to raise an exception anyways.
+    state_correct = internal.comment.decode(comment(ea, repeatable=repeatable))
+    state_wrong = internal.comment.decode(comment(ea, repeatable=not repeatable))
+    state_runtime = internal.comment.decode(function.comment(ea, repeatable=True))
```

Comment:
this call to `function.comment` needs to be guarded in case we're not in a function.